src/Common/src/System/IO/ReadOnlyMemoryStream.cs

```diff
@@ -0,0 +1,154 @@
+    internal sealed class ReadOnlyMemoryStream : Stream
+    {
+        private readonly ReadOnlyMemory<byte> _content;
+        private int _position;
+
+        public ReadOnlyMemoryStream(ReadOnlyMemory<byte> content)
+        {
+            _content = content;
+        }
+
+        public override bool CanRead => true;
+        public override bool CanSeek => true;
+        public override bool CanWrite => false;
+
+        public override long Length => _content.Length;
+
+        public override long Position
+        {
+            get => _position;
+            set
+            {
+                if (value < 0 || value > int.MaxValue)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+                _position = (int)value;
+            }
+        }
+        
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            long pos =
+                origin == SeekOrigin.Begin ? offset :
+                origin == SeekOrigin.Current ? _position + offset :
+                origin == SeekOrigin.End ? _content.Length + offset :
+                throw new ArgumentOutOfRangeException(nameof(origin));
+
+            if (pos > int.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            else if (pos < 0)
+            {
+                throw new IOException(SR.IO_SeekBeforeBegin);
+            }
+
+            _position = (int)pos;
+            return _position;
+        }
+
+        public override int ReadByte()
+        {
+            ReadOnlySpan<byte> s = _content.Span;
+            return _position < s.Length ? s[_position++] : -1;
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            ValidateReadArrayArguments(buffer, offset, count);
+            return Read(new Span<byte>(buffer, offset, count));
+        }
+
+        public override int Read(Span<byte> destination)
+        {
+            int remaining = _content.Length - _position;
+
+            if (remaining <= 0 || destination.Length == 0)
+            {
+                return 0;
+            }
+            else if (remaining <= destination.Length)
+            {
+                _content.Span.Slice(_position).CopyTo(destination);
+                _position = _content.Length;
+                return remaining;
+            }
+            else
+            {
+                _content.Span.Slice(_position, destination.Length).CopyTo(destination);
+                _position += destination.Length;
+                return destination.Length;
+            }
+        }
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            ValidateReadArrayArguments(buffer, offset, count);
+            return cancellationToken.CanBeCanceled ?
```

Comment:
I don't understand this... should this be IsCancellationRequested instead of CanBeCancelled?  (and similarly below)

Reply:
Oops... yes.